### ISSUE TITLE
Cannot use CoreClient.Events to create events

### ISSUE THREAD

**Issue by mrogers950** [openshift/origin#16587]
We added a v1.EventInterface parameter to NewServiceAccountOAuthClientGetter() for the SA OAuth events. ~~Callers can't use CoreClient for this and have to explicitly set up a CoreV1Client since CoreClient does not have this interface for some reason. Is this expected or should we be able to use CoreClient?
See here for what we are doing at the moment:~~ https://github.com/openshift/origin/blob/25b1ed1ca60bd207680eada7af8bf622e8c89099/pkg/oauth/apiserver/apiserver.go#L125-L137

**Comment by enj** [comment]
So I forget how we were using it earlier without compile errors, but there was a call to `coreClient.Events("")`.  When that interface was used to create events, the server would respond with something like "no type is registered".  So the question is not about which interface to use with the event sink, but more so about using `coreClient.Events("").Create` vs. `coreV1Client.Events("").Create`.  I would assume both would work in creating an event.

**Comment by enj** [comment]
So I ran this to see if I could recreate the original issue I saw:

```go
go func() {
	for {
		a := &v1.Event{}
		a.Name = "v1" + uuid.New()
		a.Namespace = "openshift"
		a.InvolvedObject.Name = "v1"
		a.InvolvedObject.Namespace = "openshift"
		_, v1err := coreV1Client.Events("openshift").Create(a)
		glog.Errorf("v1err: %#v", v1err)

		b := &api.Event{}
		b.Name = "v0" + uuid.New()
		b.Namespace = "openshift"
		b.InvolvedObject.Name = "v0"
		b.InvolvedObject.Namespace = "openshift"
		_, v0err := coreClient.Events("openshift").Create(b)
		glog.Errorf("v0err: %#v", v0err)

		time.Sleep(3 * time.Second)
	}
}()
```

But those both work.  @deads2k has anything changed in the last month-ish with type registration or the like?  If not I am just going to assume we were calling something incorrectly.

**Comment by enj** [comment]
@deads2k a simple answer would suffice 😄

**Comment by deads2k** [comment]
> But those both work. @deads2k has anything changed in the last month-ish with type registration or the like? If not I am just going to assume we were calling something incorrectly.

With two copies of kinds things went weird.  This was resolved in kube 1.8.

**Comment by enj** [comment]
Thanks.
